pkg/scraper/stashbox/stash_box.go

```diff
@@ -66,8 +66,79 @@ func (c Client) QueryStashBoxScene(queryStr string) ([]*models.ScrapedScene, err
 }
 
 // FindStashBoxScenesByFingerprints queries stash-box for scenes using every
-// scene's MD5/OSHASH checksum, or PHash
-func (c Client) FindStashBoxScenesByFingerprints(sceneIDs []string) ([]*models.ScrapedScene, error) {
+// scene's MD5/OSHASH checksum, or PHash, and returns results in the same order
+// as the input slice.
+func (c Client) FindStashBoxScenesByFingerprints(sceneIDs []string) ([][]*models.ScrapedScene, error) {
+	ids, err := utils.StringSliceToIntSlice(sceneIDs)
+	if err != nil {
+		return nil, err
+	}
+
+	var fingerprints []string
+	// map fingerprints to their scene index
+	fpToScene := make(map[string][]int)
+
+	if err := c.txnManager.WithReadTxn(context.TODO(), func(r models.ReaderRepository) error {
+		qb := r.Scene()
+
+		for index, sceneID := range ids {
+			scene, err := qb.Find(sceneID)
+			if err != nil {
+				return err
+			}
+
+			if scene == nil {
+				return fmt.Errorf("scene with id %d not found", sceneID)
+			}
+
+			if scene.Checksum.Valid {
+				fingerprints = append(fingerprints, scene.Checksum.String)
+				fpToScene[scene.Checksum.String] = append(fpToScene[scene.Checksum.String], index)
+			}
+
+			if scene.OSHash.Valid {
+				fingerprints = append(fingerprints, scene.OSHash.String)
+				fpToScene[scene.OSHash.String] = append(fpToScene[scene.OSHash.String], index)
+			}
+
+			if scene.Phash.Valid {
+				phashStr := utils.PhashToString(scene.Phash.Int64)
+				fingerprints = append(fingerprints, phashStr)
+				fpToScene[scene.OSHash.String] = append(fpToScene[phashStr], index)
```

Comment:
I think it should be 
```golang
fpToScene[phashStr]  = append(fpToScene[phashStr], index)
```
here instead.
it should fix the above issue, edited this from my phone so couldn't actually test it

Reply:
Thanks, this looks to have fixed the issue based on my testing.